code/modules/mapping/map_template.dm

```diff
@@ -101,15 +106,15 @@
 	var/y = round((world.maxy - height) * 0.5) + 1
 
 	var/datum/space_level/level = SSmapping.add_new_zlevel(name, list(ZTRAIT_AWAY = TRUE))
-	var/datum/parsed_map/parsed = load_map(file(mappath), x, y, level.z_value, no_changeturf=(SSatoms.initialized == INITIALIZATION_INSSATOMS), placeOnTop=TRUE)
+	var/datum/parsed_map/parsed = load_map(file(mappath), x, y, level.z_value, no_changeturf=(SSatoms.initialized == INITIALIZATION_INSSATOMS), placeOnTop=should_place_on_top)
 	var/list/bounds = parsed.bounds
 	if(!bounds)
 		return FALSE
 
 	repopulate_sorted_areas()
 
 	//initialize things that are normally initialized after map load
-	parsed.initTemplateBounds(src)
+	initTemplateBounds(bounds)
```

Comment:
These won't be offset bounds, are you sure this is correct.

Reply:
/// Offset bounds. Same as parsed_bounds until load().
	var/list/bounds

it says its offset bounds and everything seems to get initialized correctly